src/app/component/Swipeable/Swipeable.component.js

```diff
@@ -0,0 +1,126 @@
+import './Swipeable.style';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const THRESHOLD_PERC = 50;
+const TRANSITION_CSS = 'transition: transform 0.15s linear;';
+const PERCENT = 100;
+
+export class Swipeable extends React.Component {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        onRightSwipe: PropTypes.func.isRequired,
+        rightSwipeText: PropTypes.string.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.swipeObjRef = React.createRef();
+        this.leftChild = React.createRef();
+        this.rightChild = React.createRef();
+
+        this.startX = null;
+        this.deltaX = null;
+        this.storedOffset = 0;
+        this.elemWidth = null;
+
+        this.onTouchStart = this.onTouchStart.bind(this);
+        this.onTouchMove = this.onTouchMove.bind(this);
+        this.onTouchEnd = this.onTouchEnd.bind(this);
+    }
```

Comment:
Theoretically, the similar logic is in `<Draggable>` component. Can you estimate the time to migrate to it? Will it worth it?